src/api/web/index.ts

```diff
@@ -78,6 +78,24 @@ export default function(config, auth, storage) {
     });
   }
 
+  // Favicon
+  const customFaviconPath = _.get(config, 'web.favicon') || '';
+  router.get('/-/mapped/favicon', function(req, res): void {
+    if (customFaviconPath) {
+      return res.redirect('/-/mapped/custom-favicon');
+    } else {
+      return res.redirect('/-/static/favicon.ico');
+    }
+  });
+  router.get('/-/mapped/custom-favicon', function(req, res): void {
+    if (fs.existsSync(customFaviconPath)) {
+      res.sendFile(customFaviconPath);
+    } else {
+      res.status(404);
```

Comment:
We have constants for this, check `src/lib/constants.ts` or use better `import { HTTP_STATUS } from "@verdaccio/commons-api";` which was created with the idea to find references much easier.